plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -420,7 +420,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
     @Override
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
     {
-        return getTableMetadata(session, ((IcebergTableHandle) table).getSchemaTableName());
+        return getTableMetadataForExistingTable(session, ((IcebergTableHandle) table).getSchemaTableName());
```

Comment:
I think we can just use info from table handle rather than reducing it back to a name and then getting table again (even though cached). some methods will need to start using table metadata instead of table.

Reply:
We don't store the full metadata in the table handle (and probably should not). Though if you have some ideas here, we can follow up separately.